### PR TITLE
Add disksUnavailable healStatus const

### PR DESCRIPTION
**Pull request by krisis**
This is one of a two part fix for #3980. `mc admin heal` needs a change to interpret `canPartiallyHeal` heal status and inform user if an object can be healed or some disks with missing object parts are offline.

## Description

`canPartiallyHeal` healStatus constant indicates that a given object needs healing but one or more of disks requiring heal are offline. This can be used by admin heal API consumers to distinguish between a successful heal and a no-op since the outdated disks were offline. In addition, heal-object and heal-upload admin APIs now return the number of disks the object/upload was healed and the number of disks that were offline during the heal operation.

## Motivation and Context

See #3980 
## How Has This Been Tested?

Manually tested the scenario described in #3980 
## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: cmd/object-api-datatypes.go [hunk 1]
```diff
@@ -57,6 +57,7 @@ const (
 	canHeal                             // Object can be healed
 	corrupted                           // Object can't be healed
 	quorumUnavailable                   // Object can't be healed until read quorum is available
+	disksUnavailable                    // Object can't be healed completely until outdated disk(s) are online.
```
Comment (vadmeste) [inline cmd/object-api-datatypes.go]: Probably a better name for this is `canPartiallyHeal`

**Review by krishnasrinivas — COMMENTED** [review]
If 1 disk needs healing and 1 disk is offline, we will still be able to heal 1 disk right? i.e this change only affects the user message?

**Comment by krisis** [comment]
>If 1 disk needs healing and 1 disk is offline, we will still be able to heal 1 disk right? i.e this change only affects the user message?

This change allows `mc` like applications to inform users whether the object will be healed completely or partially. In addition to this, I will be changing heal-object admin API to return number of disks healed and number of disks offline during heal-object. This is to help `mc` to inform users the specifics of the heal operation.

**Review by krishnasrinivas — APPROVED** [review]

File: cmd/object-api-datatypes.go (hunk 1 above)
Comment (harshavardhana) [reply cmd/object-api-datatypes.go]: any comments on this @krisis ?

Comment (krisis) [reply cmd/object-api-datatypes.go]: I have one more change to this PR where I will address this comment.

**Pull request merged**